### PR TITLE
autoconf_archive: build needs perl

### PR DESCRIPTION
**Pull request by mtl1979**

### CODE REVIEW

**Comment by Begasus** [comment]
Build is fine without it, looking at Gentoo it doesn't mention this requirement?

https://packages.gentoo.org/packages/dev-build/autoconf-archive/dependencies

**Comment by mtl1979** [comment]
> Build is fine without it, looking at Gentoo it doesn't mention this requirement?
> 
> https://packages.gentoo.org/packages/dev-build/autoconf-archive/dependencies

Breaks here on 32-bit Haiku...

**Comment by jmairboeck** [comment]
The [Readme file](http://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=blob;f=README-maint;h=686f441623793b389b7ef2da7d80a9c148592e33;hb=HEAD#l4) also doesn't mention perl.

**Comment by Begasus** [comment]
Build is broken indeed on 32bit, can't find /bin/perl
@jmairboeck couldn't find a reference in the source also, but does indeed seem to need it.

Bump version too while in the process @mtl1979 ?

**Comment by mtl1979** [comment]
> Build is broken indeed on 32bit, can't find /bin/perl @jmairboeck couldn't find a reference in the source also, but does indeed seem to need it.
> 
> Bump version too while in the process @mtl1979 ?

Just need to check what's the latest usable version... Came back from grocery store already ;)...

**Comment by Begasus** [comment]
Latest  version builds ok on 32bit (running autoreconf on it (doesn't need perl then, only needs aclocal and autoreconf added)
PS, no tests provided.

**Comment by Begasus** [comment]
Checked with  axel_x86, no problems with newer version.

**Comment by mtl1979** [comment]
> Latest version builds ok on 32bit (running autoreconf on it (doesn't need perl then, only needs aclocal and autoreconf added) PS, no tests provided.

This comes to which way is less likely to break things... I try to avoid recreating configure script as that can cause issues when required macros are deprecated or (new) dependencies need to be downgraded.

**Comment by Begasus** [comment]
Guess it's a chicken and egg issue :) If it's fine with @jmairboeck this can be merged.

**Comment by mtl1979** [comment]
> Guess it's a chicken and egg issue :) If it's fine with @jmairboeck this can be merged.

More like raspberry or strawberry kind of thing... or apples and oranges like other people like to say...

**Comment by jmairboeck** [comment]
I didn't check it yet.

I usually prefer running the provided configure script directly if it correctly supports Haiku (which normally is no problem nowadays) and there is no patching required for the build system (configure.ac, Makefile.am etc.). If the build system requires patches, I think it is better to patch the source files and regenerate the build files instead of patching the generated ones.

**Comment by jmairboeck** [comment]
One other thing to consider when adding build dependencies is to avoid circular dependencies (i.e. if building perl required autoconf_archive directly or indirectly). But isn't the case here.

**Comment by jmairboeck** [comment]
Ah, I know what the problem is: autoconf_archive doesn't need perl directly, but the legacy gcc2 version of makeinfo does, and doesn't declare that requirement.

I would suggest using the newer makeinfo instead.

File: sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe [hunk 1]
```diff
@@ -27,6 +27,7 @@ BUILD_PREREQUIRES="
 	cmd:cmp
 	cmd:make
 	cmd:makeinfo
+	cmd:perl
```
Comment (jmairboeck) [inline sys-devel/autoconf_archive/autoconf_archive-2024.10.16.recipe:30]:
```suggestion
	cmd:makeinfo >= 7
```

**Comment by jmairboeck** [comment]
I added the missing requirement to texinfo 6 in a1f91f3.

**Pull request merged**